### ISSUE TITLE
Missing ErrorCode.HDN_ExpressionIsProbablyNeverNull diagnostics

### ISSUE THREAD

**Issue by AlekseyTs** [dotnet/roslyn#29890]
```
        [Fact]
        public void AnonymousTypes_06()
        {
            var source =
@"class C
{
    static void F(string x, string y)
    {
        x = new { x, y }.x ?? x;
        y = new { x, y = y }.y ?? y;
    }
}";
            var comp = CreateCompilation(new[] { source, NonNullTypesTrue, NonNullTypesAttributesDefinition });
            // PROTOTYPE(NullableReferenceTypes): Should report ErrorCode.HDN_ExpressionIsProbablyNeverNull.
            // See comment in DataFlowPass.VisitAnonymousObjectCreationExpression.
            comp.VerifyDiagnostics();
        }
```

**Comment by jcouv** [comment]
Relates to https://github.com/dotnet/roslyn/issues/24018

**Comment by gafter** [comment]
At LDM today we confirmed that we are dropping the HDN diagnostics.
